### DISCUSSION TITLE
strange problem with convert and IM 6.9

### DISCUSSION THREAD

**Discussion by yymsc** [ImageMagick/ImageMagick#5582] [Digital Image Processing] [answered]
Hi,
I need to work with ImageMagick 6.9 due to some legacy application, and now I'm running into a strange problem when I try to convert a tif image (black/white with LZW compression) to bmp using the following simple command:
`convert test.tif out.bmp`

Using an older IM version (6.9.12-**25** Q16-HDRI x64) everything works fine and the resulting image looks good.
However, when I use the latest 6.9 version (6.9.12-**63** Q16-HDRI x64) the result looks really strange with color pixels:
![strange](https://user-images.githubusercontent.com/4045703/191516631-f4ecbc2e-d432-4433-bf4b-4a8a2212c9a7.jpg)
[testfiles.zip](https://github.com/ImageMagick/ImageMagick/files/9618041/testfiles.zip)

Anyone has an idea why this happens with latest 6.9.12-63? 

thx
Matt

**Comment by fmw42** [comment]
When I convert your tiff on IM 6.9.12-63 Q16 Mac OSX Monterey, I get a different result

TIFF (converted to png for viewing here). **<--- But this looks different from the original tif**

![test_tiff](https://user-images.githubusercontent.com/13458218/191542856-74157682-0d99-4afc-8ed7-19145436d3bb.png)

`convert test.tif test.bmp`

BMP  (converted to png for viewing here)

![test_bmp](https://user-images.githubusercontent.com/13458218/191540705-fb552b6d-7825-4325-8d5c-22035f05f5fa.png)

Looking at your input and output images using identify, seems to indicate that they are identical (and are actually RGBA, not grayscale or binary)

However, if I extract the base image and the alpha channel, it looks like your BMP result is showing just the base image for you and your tif is showing the alpha channel only.

Base:

![test_bmp_aoff](https://user-images.githubusercontent.com/13458218/191541177-14a9bfd7-21cd-4481-a8d8-bb64feeead5b.png)


Alpha:

![test_bmp_alpha](https://user-images.githubusercontent.com/13458218/191541220-ce21bc72-f938-4cb6-889d-f3eefd6c8a2b.png)

**So I think there is something odd with your input TIF**

**If I convert from tif to tif, the result is different**

`convert test.tif test2.tif`

converted to PNG for viewing here

![test2_tif](https://user-images.githubusercontent.com/13458218/191544331-1c38cf4d-bdf0-45f2-bf95-9c49e3339e73.png)

[test2.tif.zip](https://github.com/ImageMagick/ImageMagick/files/9618032/test2.tif.zip)

**Comment by snibgo** [comment, marked as answer]
> So I think there is something odd with your input TIF

Yes. Exiftool and IM agree that the input, test.tif, has *unassociated* alpha. This means that alpha does *not* represent opacity.

Old versions of IM, and other software, ignore this, and always assume that alpha *does* represent opacity.

Current version of IM recognises that unassociated alpha is "just another channel", so it doesn't make pixels transparent.

If alpha should represent opacity, then do this:
```
magick test.tif -alpha associate out.png
```

**Reply by fmw42** [reply]
Thanks, snibgo.  I did not know about that change. Makes sense now.
